### DISCUSSION TITLE
Webgazer query

### DISCUSSION THREAD

**Discussion by SylvainUQTR** [jspsych/jsPsych#2260] [Q&A]
Hello all,

I wish to display estimated gaze points in trials during development of an experiment. However, several attempts have failed.

The trial is of type jsPsychHtmlKeyboardResponse.

Adding the following in the trial definition does not crash the script (unlike other attempts):
function () {
		    jsPsych.extensions.webgazer.showPredictions()
		  },

However, it draws a single gaze point, that's it. I have tried with ...webgazer.resume() as well just before in the function, same outcome:
function () {
		  jsPsych.extensions.webgazer.resume()  
                  jsPsych.extensions.webgazer.showPredictions()
		  },

Anyone has working code snippets (or a whole task) to show/share where/how this works?

The trial works otherwise and eye data is correctly recorded. I'd just like to see it live!

Thanks!

**Comment by TzviCohen** [comment]
Hi @SylvainUQTR , did you find a solution?
I face the same issue.

**Reply by SylvainUQTR** [reply]
Alas no. But I have set it aside for now (it is a side project) so when I get back to it, I will make sure to report on the issue if I succeed!

**Comment by brkfst42** [comment]
Encountering this issue in an experiment on Pavlovia. Would appreciate it if anyone has insights to offer. 

In my case: During calibration and validation, WebGazer is only detecting gaze in the center of the screen. When using the video-keyboard-response plugin and WebGazer, WebGazer is detecting only 1 set of x, y coordinates throughout 10-second videos. 

Using: 
`extension-webgazer@1.0.2`
`jspsych-7-pavlovia-2022.1.1.js`

**Reply by brkfst42** [reply]
How I solved this for my purpose: Using Chrome, spatial resolution is much better compared to Firefox and Safari on Mac (MacOS 12.6.3). The only-detecting-one-location issue took place on Firefox and when I tested wearing glasses. 
Therefore, I will also limit recruitment to participants who either have no need for glasses or wear contact lenses.
